python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1380 @@
+    nPixBorderGainCalc = pexConfig.Field(
+        dtype=int,
+        doc="The number of border pixels to exclude when calculating the gain",
+        default=10
+    )
+    nPixBorderXCorr = pexConfig.Field(
+        dtype=int,
+        doc="The number of border pixels to exclude when calculating the cross-correlation and kernel",
+        default=10
+    )
+    biasCorr = pexConfig.Field(
+        dtype=float,
+        doc="An empirically determined correction factor, used to correct for the sigma-clipping of" +
+        " a non-Gaussian distribution. Post DM-15277, code will exist here to calulate appropriate values",
+        default=0.9241
+    )
+    backgroundBinSize = pexConfig.Field(
+        dtype=int,
+        doc="Size of the background bins",
+        default=128
+    )
+    fixPtcThroughOrigin = pexConfig.Field(
+        dtype=bool,
+        doc="Contrain the fit of the PTC to go through the origin?",
+        default=True
+    )
+    level = pexConfig.ChoiceField(
+        doc="The level at which to calculate kernels",
+        dtype=str, default="DETECTOR",
+        allowed={
+            "AMP": "Every amplifier treated separately",
+            "DETECTOR": "One kernel per detector",
+        }
+    )
+    backgroundWarnLevel = pexConfig.Field(
+        dtype=float,
+        doc="Log warnings if the mean of the fitted background is found to be above this level after " +
+        "differencing image pair.",
+        default=0.1
+    )
+
+
+class MakeBrighterFatterKernelTaskRunner(pipeBase.TaskRunner):
+    """Subclass of TaskRunner for the MakeBrighterFatterKernelTask.
+
+    This transforms the processed arguments generated by the ArgumentParser
+    into the arguments expected by makeBrighterFatterKernelTask.run().
+
+    makeBrighterFatterKernelTask.run() takes a two arguments,
+    one of which is the dataRef (as usual), and the other is the list
+    of visitPairs, in the form of a list of tuples.
+    This list is supplied on the command line as documented,
+    and this class parses that, and passes the parsed version
+    to the run() method.
+
+    See pipeBase.TaskRunner for more information.
+    """
+
+    @staticmethod
+    def getTargetList(parsedCmd, **kwargs):
+        """Parse the visit list and pass through explicitly."""
+        visitPairs = [(int(visit.split(",")[0]), int(visit.split(",")[1])) for visit in parsedCmd.visitPairs]
+        return pipeBase.TaskRunner.getTargetList(parsedCmd, visitPairs=visitPairs, **kwargs)
+
+
+class BrighterFatterKernelTaskDataIdContainer(pipeBase.DataIdContainer):
+    """A DataIdContainer for the MakeBrighterFatterKernelTask."""
+
+    def makeDataRefList(self, namespace):
+        """Compute refList based on idList.
+
+        This method must be defined as the dataset does not exist before this
+        task is run.
+
+        Parameters
+        ----------
+        namespace
+            Results of parsing command-line
+            (with ``butler`` and ``log`` elements).
```

Comment:
I think you are saying that the "butler" and "log" elements are required. If so, it would be nice to say so explicitly, e.g. as per the original class (though you could replace "set" with "present"):
```
        namespace : `argparse.Namespace`
            Results of parsing command-line. The ``butler`` and ``log``
            elements must be set.
```

Reply:
Actually, I wasn't trying to say anything really, just following the original docs I think. It now just reads `Results of parsing command-line.`.